pkg/api/config/v1alpha1/types.go

```diff
@@ -0,0 +1,83 @@
+/*
+The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
```

Comment:
I noticed there's no internal type in this PR. Have you thought about how to generate conversions for your ComponentConfigs when you introduce new versions?

Reply:
This is definitely an area that I haven't worked out entirely yet. I'm slightly resting on the fact that Manager is still using `Options` as it's core object. My current thinking is not to handle these cases just yet and to follow up this initial implementation with trying to change the Manager to use an internal type if possible and the idea would be to implement the conversions then. 

Based on your experience would that work to punt that to the next phase? Speaking of which I should probably have an overall tracking issue for all of this work.

Reply:
You don't need to deal with it until you're going to exit alpha. But at that point it must be dealt with.